File: app/src/main/java/com/sliebald/cula/data/database/pojos/LessonKnowledgeLevel.java
package com.sliebald.cula.data.database.pojos;

import androidx.annotation.NonNull;
import androidx.room.ColumnInfo;


/**
 * Simple POJO class that holds a lesson and the average of Knowledge Levels of all words in that
 * lesson.
 */
public class LessonKnowledgeLevel {

    /**
     * Lesson Name.
     */
    @ColumnInfo(name = "lessonName")
    public String lessonName;

    /**
     * Average of KnowledgeLevel of all
     * {@link com.sliebald.cula.data.database.entities.LibraryEntry}s assigned to the
     * lesson.
     */
    @ColumnInfo(name = "average")
    public double average;

    @NonNull
    @Override
    public String toString() {
        return "LessonKnowledgeLevel{" +
                "lessonName='" + lessonName + '\'' +
                ", average=" + average +
                '}';
    }
}

File: app/src/main/java/com/sliebald/cula/data/database/pojos/LessonKnowledgeLevel.kt
package com.sliebald.cula.data.database.pojos

/**
 * Simple POJO class that holds a lesson and the average of Knowledge Levels of all words in that
 * lesson.
 *
 * @param lessonName Lesson Name.
 * @param average Average of KnowledgeLevel of all [com.sliebald.cula.data.database.entities.LibraryEntry]s assigned to the lesson.
 */
data class LessonKnowledgeLevel(var lessonName: String? = null, var average: Double = 0.0)